Modules/timemodule.c

```diff
@@ -1135,7 +1137,9 @@ time_tzset(PyObject *self, PyObject *unused)
         return NULL;
     }
 
+#if !defined(MS_WINDOWS) || defined(MS_WINDOWS_DESKTOP)
```

Comment:
```suggestion
#if !defined(MS_WINDOWS) || defined(MS_WINDOWS_DESKTOP) || defined(MS_WINDOWS_SYSTEM)
```

Since there are multiple checks for the CRT's extended desktop partition that includes the system partition, maybe we should define `MS_WINDOWS_CRT_DESKTOP`.

Reply:
I added `MS_WINDOWS_CRT_DESKTOP` and use it for this detection now.

Reply:
I'd prefer `MS_WINDOWS_DESKTOP_OR_SERVER` (and wouldn't be opposed to `MS_WINDOWS_SERVER` instead of system). 

It's about being able to understand what's going on without having to look up what the term means.

Reply:
I do not particularly care about this name. Just let me know which name to update it to.

Reply:
Keep them separate. If we start making single variables for every combination then it'll never stop.

If it's easy to rename to `MS_WINDOWS_SERVER` then do that - I think it'll save some explanations in the future.

Reply:
sure I will revert to the old state

Reply:
I had originally meant for `MS_WINDOWS_CRT_DESKTOP` to be based on the CRT macro `_CRT_USE_WINAPI_FAMILY_DESKTOP_APP` in "corecrt.h". But it's not actually documented for use. From "corecrt.h":

```c
#ifndef _CRT_USE_WINAPI_FAMILY_DESKTOP_APP
    #ifdef WINAPI_FAMILY
        #include <winapifamily.h>
        #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
            #define _CRT_USE_WINAPI_FAMILY_DESKTOP_APP
        #else
            #ifdef WINAPI_FAMILY_PHONE_APP
                #if WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP
                    #define _CRT_USE_WINAPI_FAMILY_PHONE_APP
                #endif
            #endif

            #ifdef WINAPI_FAMILY_GAMES
                #if WINAPI_FAMILY == WINAPI_FAMILY_GAMES
                    #define _CRT_USE_WINAPI_FAMILY_GAMES
                #endif
            #endif
        #endif
    #else
        #define _CRT_USE_WINAPI_FAMILY_DESKTOP_APP
    #endif
#endif
```